### DISCUSSION TITLE
RAM usage calculation.

### DISCUSSION THREAD

**Discussion by ghost** [aristocratos/btop#165] [Q&A] [answered]
Can someone explain to me how ram usage is calculated? 
I noticed a difference between htop and btop scores and hence this question.

**Comment by aristocratos** [comment, marked as answer]
@jokerekv2

Hi, I'm assuming you mean for linux?

Memory stats are read from `/proc/meminfo`.
But since `/proc/meminfo` doesn't provide a "used" metric, it's calculated from `total memory` - `available memory`.
So "used memory" = "non available memory".
Haven't look at how `htop` calculates this but the current calculation is what made most sense to me.

The code in question:
https://github.com/aristocratos/btop/blob/e2668c60f6e3a7ccb698d6216462d59703976c82/src/linux/btop_collect.cpp#L729-L766
